### PR TITLE
Angular app restructure

### PR DESCRIPTION
**Pull request by deepu105**
as discussed in #2271

### CODE REVIEW

File: app/templates/Gruntfile.js [hunk 1]
```diff
@@ -231,11 +231,11 @@ module.exports = function (grunt) {
         ngtemplates:    {
             dist: {
                 cwd: 'src/main/webapp',
-                src: ['scripts/app/**/*.html', 'scripts/components/**/*.html',],
+                src: ['app/app/**/*.html', 'app/components/**/*.html',],
```
Comment (andidev) [inline app/templates/Gruntfile.js]: Shouldn't one app be removed from app/app

File: app/templates/gulpfile.js [hunk 2]
```diff
@@ -270,7 +270,7 @@ gulp.task('ngconstant:dev', function() {
             VERSION: <% if(buildTool == 'maven') { %>parseVersionFromPomXml()<% } else { %>parseVersionFromBuildGradle()<% } %>
         }
     })
-    .pipe(gulp.dest(yeoman.app + 'scripts/app/'));
+    .pipe(gulp.dest(yeoman.app + 'app/app/'));
```
Comment (andidev) [inline app/templates/gulpfile.js]: Here as well

File: app/templates/gulpfile.js [hunk 3]
```diff
@@ -286,11 +286,11 @@ gulp.task('ngconstant:prod', function() {
             VERSION: <% if(buildTool == 'maven') { %>parseVersionFromPomXml()<% } else { %>parseVersionFromBuildGradle()<% } %>
         }
     })
-    .pipe(gulp.dest(yeoman.tmp + 'scripts/app/'));
+    .pipe(gulp.dest(yeoman.tmp + 'app/app/'));
```
Comment (andidev) [inline app/templates/gulpfile.js]: And here

Comment (deepu105) [inline app/templates/gulpfile.js]: yes u are rite, fixed

**Comment by deepu105** [comment]
Im also removing all the cleanup commands as it wouldn't make sense to have for all these files

File: entity/USAGE [hunk 4]
```diff
@@ -10,12 +10,12 @@ Example:
         src/main/java/package/repository/FooRepository.java
         src/main/java/package/web/rest/FooResource.java
         src/main/resources/config/liquibase/changelog/20150128232313_added_entity_Foo.xml
-        src/main/webapp/scripts/app/entities/foo/foos.html
-        src/main/webapp/scripts/app/entities/foo/foo-detail.html
-        src/main/webapp/scripts/app/entities/foo/foo.js
-        src/main/webapp/scripts/app/entities/foo/foo.controller.js
-        src/main/webapp/scripts/app/entities/foo/foo-detail.controller.js
-        src/main/webapp/scripts/components/entities/foo/foo.service.js
+        src/main/webapp/app/app/entities/foo/foos.html
```
Comment (deepu105) [inline entity/USAGE]: Need to update this

**Comment by deepu105** [comment]
@jdubois its ready now

**Comment by deepu105** [comment]
I have fixed all the errors i could find @jdubois you can merge this if you are ok

**Comment by jdubois** [comment]
I have tested and it looks good, it's so big that it's hard to review.

- Main concern is that we have things like /src/main/webapp/app/account at the same level as the "service" and "component" folders -> is this supposed to be like this in John Papa's style guide?
- I have tested the production build, we need to check if the production filters work OK, but I don't see a big issue here

**Comment by deepu105** [comment]
yes its according to the style here https://github.com/johnpapa/angular-styleguide#application-structure-lift-principle
and https://github.com/johnpapa/angular-styleguide#application-structure 

so it comes down to this
```
src
|__main
      |__webapp
           |__scss
           |__content
                |__css
                |__images
           |__i18n
           |__swagger-ui
           |__bower_components 
           |__app
                |__app.module.js
                |__app.config.js
                |__app.constants.js
                |__components
                     |__alert
                     |__form
                     |__interceptor
                     |__language
                     |__util
                |__services
                     |__admin
                     |__auth
                     |__user
                     |__entities
                |__layouts
                     |__navbar
                     |__error
                     |__login
               |__account
               |__admin
               |__entities
               |__home        
           |__index.html
           ....

```

**Comment by deepu105** [comment]
I think we can move the admin services to corresponding admin module folder as well

**Comment by vivekmore** [comment]
+1 @deepu105

**Pull request merged**